robot/Cumulus/resources/NPSP.py

```diff
@@ -51,7 +51,8 @@ def _init_locators(self):
             response = client._call_salesforce(
                 'GET', 'https://{}/services/data'.format(client.sf_instance))
             latest_api_version = float(response.json()[-1]['version'])
-        except RobotNotRunningError:
+            assert latest_api_version  in locators_by_api_version
+        except (RobotNotRunningError, AssertionError):
```

Comment:
I'd prefer an explicit

    if latest_api_version not in locators_by_api_version:
        latest_api_version = max(locators_by_api_version)

`assert` anywhere other than in pytest always makes me spend extra brain cycles convincing myself no one is going to run python with -O. But more to the point, I don't want to accidentally mask other assertions that could exist in the code invoked by _call_salesforce.

We should also probably log something in this case. Actually it wouldn't hurt to always log which version of the locators was chosen.

Reply:
What logger is appropriate in NPSP.py? A new one?

Reply:
I would use self.builtin.log() which I see used elsewhere in the module